llvm/lib/Target/RISCV/RISCVTargetTransformInfo.cpp

```diff
@@ -1060,15 +1060,28 @@ RISCVTTIImpl::getIntrinsicInstrCost(const IntrinsicCostAttributes &ICA,
     auto NVT = LT.second;
     if (ST->hasVInstructions() && LT.second.isVector()) {
       SmallVector<unsigned, 3> Opcodes;
-      // f16 with zvfhmin and bf16 with zvfbfmin
+      // f16 with zvfhmin and bf16 with zvfbfmin and the type of nxv32[b]f16
+      // will be spilt.
       if (LT.second.getVectorElementType() == MVT::bf16) {
-        Opcodes = {RISCV::VFWCVTBF16_F_F_V, RISCV::VFSQRT_V,
-                   RISCV::VFNCVTBF16_F_F_W};
-        NVT = TLI->getTypeToPromoteTo(ISD::FSQRT, NVT);
+        if (LT.second == MVT::nxv32bf16) {
+          Opcodes = {RISCV::VFWCVTBF16_F_F_V, RISCV::VFWCVTBF16_F_F_V,
+                     RISCV::VFSQRT_V,         RISCV::VFSQRT_V,
+                     RISCV::VFNCVTBF16_F_F_W, RISCV::VFNCVTBF16_F_F_W};
+        } else {
+          Opcodes = {RISCV::VFWCVTBF16_F_F_V, RISCV::VFSQRT_V,
+                     RISCV::VFNCVTBF16_F_F_W};
+          NVT = TLI->getTypeToPromoteTo(ISD::FSQRT, NVT);
+        }
       } else if (LT.second.getVectorElementType() == MVT::f16 &&
                  !ST->hasVInstructionsF16()) {
-        Opcodes = {RISCV::VFWCVT_F_F_V, RISCV::VFSQRT_V, RISCV::VFNCVT_F_F_W};
-        NVT = TLI->getTypeToPromoteTo(ISD::FSQRT, NVT);
+        if (LT.second == MVT::nxv32f16) {
+          Opcodes = {RISCV::VFWCVT_F_F_V, RISCV::VFWCVT_F_F_V,
+                     RISCV::VFSQRT_V,     RISCV::VFSQRT_V,
```

Comment:
NVT needs to be changed to v16f32.

Reply:
@LiqinWeng Are you trying to compute the cost for ISAs without f16 support as (vfwcvt, nxv*f16) + (vfsqrt, nxv*f32) + (vfncvt, nxv*f16)? From the current code, it doesn’t seem like that’s what’s being implemented.

Reply:
Under what circumstances will it occur:  (vfwcvt, nxvf16) + (vfsqrt, nxvf32) + (vfncvt, nxv*f16)？？？I don't quite understand ‘ISAs without f16’？ do you means the mattr only support zvfh？？？